### PR TITLE
Remove outdated sdkbase directory and docs

### LINKED ISSUE #351: Outdated 'sdkbase' directory
The contents of the `sdkbase` directory have been replaced by https://github.com/kbase/sdkbase2. The `sdkbase` directory should be removed, and references to it in the SDK codebase and docs updated to refer to the current GitHub repo and Docker images. The Dockerfile generated in new projects already uses the more up-to-date SDK base image, so having the remnants of the old system hanging around is just confusing.

### PR DESCRIPTION
**Pull request by ialarmedalien**
The `sdkbase` directory is no longer used as new KBase apps are built on [`sdkbase2`](https://github.com/kbase/sdkbase2). This PR removes the old sdkbase directory and mentions of it in the documentation.

Closes https://github.com/kbase/kb_sdk/issues/351.

### CODE REVIEW

File: doc/building_sdk.md [hunk 1]
```diff
@@ -1,34 +1,30 @@
...
 
 Pull dependencies and configure the SDK:
 
     cd kb_sdk
     make bin
 
-Download the local KBase SDK base Docker image:
-
-    docker pull kbase/sdkbase2:latest
-
```
Comment (ialarmedalien) [inline doc/building_sdk.md:23]: This is not the correct image for python apps, which would presumably be what the majority of people will be building. The correct image will be pulled automatically when building a new app, so it's easier to omit this step completely.

**Comment by MrCreosote** [comment]
Where does the SDK base image Dockerfile live these days?

Edit: 
I guess I have two questions:
1) Where does the Dockerfile for the app base image(s) live
2) Where does the Dockerfile for the dockerized version of kb-sdk live

File: doc/building_sdk.md (hunk 1 above)
Comment (sychan) [reply doc/building_sdk.md:23]: So have we tried to rebuild some images with this setup? I'm somewhat leery when we remove a while directory.

**Comment by ialarmedalien** [comment]
@MrCreosote:

1. The app base images are in https://github.com/kbase/sdkbase2; there are two branches, `python`, for py3 apps, and `master`, for perl and java apps (and py2). When you use the SDK to create an app, the dockerfile will be based on one of those two images.
2. Are you referring to the Dockerfile for kb-sdk itself - i.e. what makes the image that you get if you run `docker pull kbase/kb-sdk`? If so, it is in this repo: https://github.com/kbase/kb_sdk/blob/develop/Dockerfile

File: doc/building_sdk.md (hunk 1 above)
Comment (ialarmedalien) [reply doc/building_sdk.md:23]: The automated tests for this repo all use this setup (i.e. ignoring the `sdkbase` directory), as does anyone who runs `docker pull kbase/kb-sdk`.

The only way that people might use the code is if they ran `make sdkbase`, which builds an app base image with the ID `kbase/kbase:sdkbase2.latest`. There were some vague/incomplete instructions in the docs about building `kb-sdk` from source about running `make sdkbase`, but I found that whenever I ran the command and then used `kb-sdk` to generate apps, the apps were on a weird base image, and some additional tests that I'd put in failed. I believe this was the precursor to the modern sdkbase images, which are now held in https://github.com/kbase/sdkbase2.

**Comment by MrCreosote** [comment]
> Are you referring to the Dockerfile for kb-sdk itself - i.e. what makes the image that you get if you run docker pull kbase/kb-sdk? If so, it is in this repo: https://github.com/kbase/kb_sdk/blob/develop/Dockerfile

Wait - so dockerized kb-sdk isn't released?

So am I right in thinking that the sdkbase directory is completely unused everywhere, based on the answers above?

**Comment by ialarmedalien** [comment]
@MrCreosote Yup! The sdkbase directory is unused.

**Comment by MrCreosote** [comment]
And dockerized kb-sdk isn't released?

**Comment by ialarmedalien** [comment]
Dockerized kb-sdk is available via `docker pull kbase/kb-sdk`.

**Comment by MrCreosote** [comment]
But the dockerfile is on the develop branch, hence not released, right?

**Comment by ialarmedalien** [comment]
This is the current status of `kb-sdk` docker images:

https://hub.docker.com/r/kbase/kb-sdk/tags

`master` (aka `latest`) has not been updated in 2+ years. The kb-sdk image tagged `develop` will soon be automatically uploaded every time there's a merge into `develop`.

**Review by sychan — APPROVED** [review]

**Pull request merged**